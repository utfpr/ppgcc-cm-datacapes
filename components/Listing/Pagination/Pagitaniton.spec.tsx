import { render } from "@testing-library/react";
import { Pagination } from "./index";

describe("Pagination components", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Pagination
        onPageChange={() => {}}
        totalCountOfRegisters={100}
        currentPage={1}
        registersPerPage={10}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should render 3 pagination items when current page is 1", () => {
    const { container } = render(
      <Pagination
        onPageChange={() => {}}
        totalCountOfRegisters={100}
        currentPage={1}
        registersPerPage={10}
      />
    );

    expect(container.querySelectorAll("button").length).toBe(3);
  });
  it("should render 5 pagination items when current page is 5", () => {
    const { container } = render(
      <Pagination
        onPageChange={() => {}}
        totalCountOfRegisters={100}
        currentPage={5}
        registersPerPage={10}
      />
    );

    expect(container.querySelectorAll("button").length).toBe(5);
  });
});
