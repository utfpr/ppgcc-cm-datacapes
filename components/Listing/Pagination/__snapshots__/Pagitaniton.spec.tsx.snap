// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination components should render correctly 1`] = `
<div>
  <div
    class="chakra-stack css-1n17tlq"
  >
    <div
      class="css-kbn7if"
    >
      <strong>
        0
      </strong>
       - 
      <strong>
        10
      </strong>
       de 
      <strong>
        100
      </strong>
    </div>
    <div
      class="chakra-stack css-1689b8b"
    >
      <button
        class="chakra-button css-1jn0x3k"
        disabled=""
        type="button"
      >
        1
      </button>
      <button
        class="chakra-button css-1ysjku8"
        type="button"
      >
        2
      </button>
      <p
        class="chakra-text css-8t3u5b"
      >
        ...
      </p>
      <button
        class="chakra-button css-1ysjku8"
        type="button"
      >
        10
      </button>
    </div>
  </div>
</div>
`;
